lang/LangPrimSource/PyrUnixPrim.cpp

```diff
@@ -147,35 +152,29 @@ static void string_popen_thread_func(struct sc_process *process)
 int prString_POpen(struct VMGlobals *g, int numArgsPushed);
 int prString_POpen(struct VMGlobals *g, int numArgsPushed)
 {
-	struct sc_process *process;
 	PyrSlot *a = g->sp - 1;
 	PyrSlot *b = g->sp;
-	int err;
 
 	if (!isKindOfSlot(a, class_string)) return errWrongType;
 
-	char *cmdline = (char*)malloc(slotRawObject(a)->size + 1);
-	err = slotStrVal(a, cmdline, slotRawObject(a)->size + 1);
-	if(err) {
```

Comment:
These error checks are not necessary because above the slot was already checked for class_string which is the same check done inside slotStrVal